src/FileIndexer.ts

```diff
@@ -1,601 +1,2459 @@
+        return false;
+    }
+
+    private visitComputedPropertyName(node: ts.ComputedPropertyName): boolean {
+        if (this.dev) console.log(`${this.indent()}• visitComputedPropertyName [${node.pos}:${node.end}]`);
+        this.addOperatorsToAllHalstead(['[]']);
+        this.continueWalk(node.expression);
+        return false;
+    }
+
+    private visitTypeParameterDeclaration(node: ts.TypeParameterDeclaration): boolean {
+        if (this.dev) console.log(`${this.indent()}• visitTypeParameterDeclaration [${node.pos}:${node.end}]`);
+        node.modifiers?.forEach((modifier) => this.continueWalk(modifier));
+        if (node.constraint) {
+            this.addOperatorsToAllHalstead(['[]']);
+            this.continueWalk(node.name);
+            this.addOperatorsToAllHalstead(['in']);
+            this.continueWalk(node.constraint);
+            return false;
+        }
+        if (node.default) {
+            this.addOperatorsToAllHalstead(['=']);
+            this.continueWalk(node.default);
+        }
+        if (node.expression) throw new Error('************************ URGENT & NEW, node.expression');
+        this.continueWalk(node.name);
+        if (this.dev) console.log(`${this.indent()}• visitTypeParameterDeclaration [${node.pos}:${node.end}] <<EXIT>>`);
+        return false;
+    }
+
+    private visitTypeReferenceNode(node: ts.TypeReferenceNode) {
+        if (this.dev) console.log(`${this.indent()}• visitTypeReferenceNode [${node.pos}:${node.end}]`);
+        this.continueWalk(node.typeName);
+        if (node.typeArguments) {
+            this.addOperatorsToAllHalstead(['<>']);
+            this.visitNodeArray(node.typeArguments, '', ',', false);
+        }
+        if (this.dev) console.log(`${this.indent()}• visitTypeReferenceNode [${node.pos}:${node.end}] <<EXIT>>`);
+        return false;
+    }
+
+    private visitVariableStatement(node: ts.VariableStatement): boolean {
+        if (this.dev) console.log(`${this.indent()}• visitVariableStatement [${node.pos}:${node.end}]`);
+        node.modifiers?.forEach((modifier) => this.continueWalk(modifier));
+        this.continueWalk(node.declarationList);
+        if (this.dev) console.log(`${this.indent()}• visitVariableStatement [${node.pos}:${node.end}] <<EXIT>>`);
+        return false;
+    }
+
+    private visitVariableDeclarationList(node: ts.VariableDeclarationList): boolean {
+        if (this.dev) console.log(`${this.indent()}• visitVariableDeclarationList [${node.pos}:${node.end}]`);
+        node.modifiers?.forEach((modifier) => this.continueWalk(modifier));
+
+        if (node.flags == ts.NodeFlags.None || node.flags & ts.NodeFlags.None) this.addOperatorsToAllHalstead(['var']);
+        if (node.flags == ts.NodeFlags.Let || node.flags & ts.NodeFlags.Let) this.addOperatorsToAllHalstead(['let']);
+        if (node.flags == ts.NodeFlags.Const || node.flags & ts.NodeFlags.Const)
+            this.addOperatorsToAllHalstead(['const']);
```

Comment:
```suggestion
        if (node.flags & ts.NodeFlags.None) this.addOperatorsToAllHalstead(['var']);
        if (node.flags & ts.NodeFlags.Let) this.addOperatorsToAllHalstead(['let']);
        if (node.flags & ts.NodeFlags.Const) this.addOperatorsToAllHalstead(['const']);
```
No need to test for equality when you have bit flags; the bitwise AND (`&`) will take care of everything for you.

Reply:
Yeah - that's what I thought!  I added this in because it was being borky!  I will mark as todo with a link to this, but I originally had what you put ;)